missing `fail` instance for Identity

When building with GHC 8.4.4, I'm getting the following error:

```
src/Language/Go/Rename.hs:86:18: error:
    • Could not deduce (MonadFail Identity)
        arising from a use of ‘fail’
      from the context: Show a
        bound by the type signature for:
                   rename_alg :: forall a (tp :: NodeType).
                                 Show a =>
                                 NodeF a (Product (Node a) (RM a)) tp -> RM a tp
        at src/Language/Go/Rename.hs:59:1-71
      or from: tp ~ 'Stmt
        bound by a pattern with constructor:
                   AssignStmt :: forall a (f :: NodeType -> *).
                                 a
                                 -> AssignType
                                 -> Maybe BinaryOp
                                 -> [f 'Expr]
                                 -> [f 'Expr]
                                 -> NodeF a f 'Stmt,
                 in an equation for ‘rename_alg’
        at src/Language/Go/Rename.hs:78:13-47
    • In the expression:
        fail "rename_alg AssignStmt: expected ident in LHS"
      In a case alternative:
          _node -> fail "rename_alg AssignStmt: expected ident in LHS"
      In the expression:
        case node of
          In (IdentExpr _x _tp _q (Ident _k name)) -> addLocal name
          _node -> fail "rename_alg AssignStmt: expected ident in LHS"
```

--- a/src/Language/Go/Rename.hs
+++ b/src/Language/Go/Rename.hs
@@ -83,7 +83,7 @@ rename_alg (AssignStmt x assign_type op lhs rhs) = RM $ do
     Define -> do
       forM_ lhs $ \(Pair node _rm) -> case node of
         In (IdentExpr _x _tp _q (Ident _k name)) -> addLocal name
-        _node -> fail "rename_alg AssignStmt: expected ident in LHS"
+        _node -> error "rename_alg AssignStmt: expected ident in LHS"
     _ -> return ()
   In <$> (AssignStmt x assign_type op <$> mapM run lhs <*> mapM run rhs)
 

